implementations/rust/ockam/ockam_command/src/enroll/oidc_service.rs

```diff
@@ -134,6 +134,33 @@ impl OidcService {
         res.json().await.map_err(|e| miette!(e).into())
     }
 
+    pub async fn wait_for_email_verification(
+        &self,
+        token: &OidcToken,
+        opts: &CommandGlobalOpts,
+    ) -> Result<UserInfo> {
+        let spinner_option = opts.terminal.progress_spinner();
+        loop {
+            let user_info = self.get_user_info(token).await?;
```

Comment:
There is a decent chance we hit a 429 `Too Many Requests`, current retry strategy go up to maximum 1 second: 
```
let retry_strategy = ExponentialBackoff::from_millis(10).take(3);
```
I think we should either relax it a little or handle 429 in a slightly different way

Reply:
right, I'll create a ticket to handle this error more gracefully.    Here we are using a fixed 5 seconds delay between attempts,   we can tweak that too (although  I didn't want to use a large max backoff  time due to not being user frienly:  we are blocking the ui until the user verify his email,   if we take too long _after_ it has verified it to respond, it will be confusing)

Reply:
https://github.com/build-trust/ockam/issues/5630